# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../musl/pkg/openvpn/openvpn-no-struct_ethhdr-redefinition.patch
# Copyright (C) 2014 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

On musl libc based systems 'struct ethhdr' is defined in <netinet/if_ether.h>
but when including <linux/if_tun.h> it pulls in <linux/if_ether.h> which is
also defining 'struct ethhdr'.

-----------------------------------------------------------------------------
/usr/include/netinet/if_ether.h:94:8: note: originally defined here
In file included from /usr/include/linux/if_tun.h:20:0
/usr/include/linux/if_ether.h:133:8: error: redefinition of 'struct ethhdr'
In file included from syshead.h:218:0
-----------------------------------------------------------------------------

--- a/src/openvpn/syshead.h	2014-04-26 02:11:53.110728493 +0200
+++ b/src/openvpn/syshead.h	2014-04-26 02:16:59.229770416 +0200
@@ -214,10 +214,6 @@
 
 #ifdef TARGET_LINUX
 
-#if defined(HAVE_NETINET_IF_ETHER_H)
-#include <netinet/if_ether.h>
-#endif
-
 #ifdef HAVE_LINUX_IF_TUN_H
 #include <linux/if_tun.h>
 #endif
